### PR TITLE
Enable WMO for Embedded targets automatically

### PR DESCRIPTION
**Pull request by rauhul**
Updates SwiftPM to enable WMO for targets with experimental-feature Embedded. Unifies the logic to enable WMO into
SwiftModuleBuildDescription.

### CODE REVIEW

**Comment by rauhul** [comment]
@swift-ci smoke test

**Comment by rauhul** [comment]
@swift-ci test

**Comment by rauhul** [comment]
@swift-ci test

**Comment by rauhul** [comment]
@swift-ci smoke test

**Comment by rauhul** [comment]
@swift-ci test windows

**Review by MaxDesiatov — APPROVED** [review]

File: Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift [hunk 1]
```diff
@@ -987,6 +984,19 @@ public final class SwiftModuleBuildDescription {
 
         return arguments
     }
+
+    /// Whether to build Swift code with whole module optimization (WMO)
+    /// enabled.
+    package var useWholeModuleOptimization: Bool {
+        if self.target.underlying.isEmbeddedSwiftTarget { return true }
```
Comment (dschaefer2) [inline Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift:991]: What does this impact if we decide we want to support debugging of embedded?

**Review by dschaefer2 — COMMENTED** [review]
Max approved this but I did have questions.

File: Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift (hunk 1 above)
Comment (rauhul) [reply Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift:991]: The embedded compilation mode _requires_ `-wmo` to be passed to swiftc when in `-Onone` and with/without `-g` so it should be orthogonal. 

Without this particular patch, `swift build -c debug` does not work with embedded targets.

Comment (dschaefer2) [reply Sources/Build/BuildDescription/SwiftModuleBuildDescription.swift:991]: Thanks!

**Pull request merged**